### PR TITLE
Port editor to Qt5

### PR DESCRIPTION
**Pull request by CounterPillow**
Some things to note:

* `#include <QtGui>` was replaced with `#include <QtWidgets>`, as this is the Qt5 thing to include now.

* GLEW had to be removed. It clashed with Qt5's insistence of doing its own stuff. It was replaced with Qt5's QOpenGLFunctions facilities.

 * Unfortunately, this means calling gl functions now requires going through an object, as Qt does not expose the functions as static functions in some namespace, but as members of a class object. OpenGL.hpp does this through the symbol `glf` in the `Tungsten::GL` namespace; it is initialised once `initOpenGL` is called.

 * Fortunately, this means there is no more platform specific code for OS X. 

* `QGLWidget` was replaced by `QOpenGLWidget`, as this is the more modern alternative.

 * All `updateGL()` calls have been replaced by `update()`

 * Since `QOpenGLWidget` renders to its own FBO and not directly to the surface any longer, src/editor/PreviewWindow.cpp, line 628 is thus needed to re-bind the proper FBO after the primitive picking.

 * the `glViewport()` call inside `PreviewWindow::paintGL()` may now be obsolete, as according to [the Qt5 documentation](http://doc.qt.io/qt-5/qopenglwidget.html#paintGL) this should be done by Qt5 for us; I haven't removed them for now though to avoid regressions before any other eyes have confirmed that my assumptions are true.

* We now use `QSurfaceFormat` instead of `QGLFormat`. It is set as the default format, which may be a bad idea, since we don't need multisampling and depth + alpha buffers for everything. I'll look into refactoring the code to avoid this, the current one is just what I could get working at the moment.

* Multisampling was reduced from 16 down to 4.

### CODE REVIEW

**Comment by CounterPillow** [comment]
I added 2 more commits, one to fix the red glClear colour I left in by accident and used for debug purposes, the other to not set the surface format as default format.

Instead of passing a format as parameter of the PreviewWindow constructor like it was initially done with QGLFormat, the constructor decides what format to use on its own, as the class itself depends on a specific OpenGL version and thus parametrising this aspect of construction does not make sense to me.

Also note that the omission of a check for the resulting profile of the context is intentional; we don't care whether we get a compatibility or a core profile context as long as it's OpenGL 3.2, but we request a core profile to let the driver know we only require the core profile functionality, which for some drivers is all they can provide with modern OpenGL.

File: CMakeLists.txt [hunk 1]
```diff
@@ -140,14 +136,15 @@ if (OPENGL_FOUND AND GLEW_FOUND AND QT4_FOUND)
...
         find_library(APPKIT NAMES AppKit) 
         target_link_libraries(editor ${APPKIT})
     endif()
 else()
-    if (NOT QT4_FOUND)
-        message(STATUS "Qt4 not found. Editor will not be built")
+    if (NOT Qt5Widgets_FOUND)
+        message(STATUS "Qt5 not found. Editor will not be built")
     endif()
     if (NOT (GLEW_FOUND AND OPENGL_FOUND))
```
Comment (tunabrain) [inline CMakeLists.txt]: Do we still need this check? At least GLEW_FOUND can probably be removed

File: src/editor/opengl/OpenGL.cpp [hunk 2]
```diff
@@ -1,16 +1,17 @@
+#include <QOpenGLWidget>
 #include "OpenGL.hpp"
 
 namespace Tungsten {
 
 namespace GL {
 
+QOpenGLFunctions_3_2_Core* glf = 0;
```
Comment (tunabrain) [inline src/editor/opengl/OpenGL.cpp]: This is very nitpicky, but can you move the * one to the right, next to glf? That's the coding style used in the rest of the project - & and * next to the variable name, not the type.

File: src/editor/opengl/OpenGL.hpp [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
-#else
-# include <GL/glew.h>
-#endif
+#include <QOpenGLFunctions_3_2_Core>
 
 namespace Tungsten {
 
 namespace GL {
 
+extern QOpenGLFunctions_3_2_Core* glf;
```
Comment (tunabrain) [inline src/editor/opengl/OpenGL.hpp]: Similar comment to above, can you put the * next to glf instead of the type?

**Comment by tunabrain** [comment]
Thanks for your work on this! This is really cool.

**Pull request merged**